pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -1100,18 +1106,45 @@ func validateNUMA(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec, con
 					field.Child("domain", "cpu", "dedicatedCpuPlacement").String(),
 					field.Child("domain", "cpu", "numa", "guestMappingPassthrough").String(),
 				),
-				Field: field.Child("domain", "cpu", "numa", "guestMappingPassthrough").String(),
+				Field: field.Child("domain", "cpu", "numa").String(),
 			})
 		}
-		if spec.Domain.Memory == nil || spec.Domain.Memory.Hugepages == nil {
-			causes = append(causes, metav1.StatusCause{
-				Type: metav1.CauseTypeFieldValueInvalid,
-				Message: fmt.Sprintf("%s must be requested when NUMA topology strategy is set in %s",
-					field.Child("domain", "memory", "hugepages").String(),
-					field.Child("domain", "cpu", "numa", "guestMappingPassthrough").String(),
-				),
-				Field: field.Child("domain", "cpu", "numa", "guestMappingPassthrough").String(),
-			})
+
+		if spec.Domain.CPU.NUMA.GuestMappingPassthrough != nil {
+			if spec.Domain.CPU.NUMA.MemoryMode != nil {
+				causes = append(causes, metav1.StatusCause{
+					Type: metav1.CauseTypeFieldValueInvalid,
+					Message: fmt.Sprintf("%s can't be set alongside %s",
+						field.Child("domain", "cpu", "numa", "guestMappingPassthrough").String(),
+						field.Child("domain", "cpu", "numa", "memory_mode").String(),
+					),
+					Field: field.Child("domain", "cpu", "numa", "guestMappingPassthrough").String(),
+				})
+			}
+			if spec.Domain.Memory == nil || spec.Domain.Memory.Hugepages == nil {
+				causes = append(causes, metav1.StatusCause{
+					Type: metav1.CauseTypeFieldValueInvalid,
+					Message: fmt.Sprintf("%s must be requested when NUMA topology strategy is set in %s",
+						field.Child("domain", "memory", "hugepages").String(),
+						field.Child("domain", "cpu", "numa", "guestMappingPassthrough").String(),
+					),
+					Field: field.Child("domain", "cpu", "numa", "guestMappingPassthrough").String(),
+				})
+			}
+		}
+
+		if spec.Domain.CPU.NUMA.MemoryMode != nil {
+			memMode := *spec.Domain.CPU.NUMA.MemoryMode
+			if valid := memoryModes[memMode]; !valid {
```

Comment:
I'm not sure, but I think that this isn't necessary because `MemoryMode` is defined to be a set of specific strings, therefore if the value found isn't one of the values of `MemoryMode`, validation would fail.

Although I've tested it with `ioThreadsPolicy` (was setting up a gibberish policy and validation failed), I don't see anything in our auto generated files that actually enforces that, so I'm not sure where this validation comes from.

@xpivarc can you keep me honest here?

Reply:
This is how Kubevirt do validation for other API types that are type aliases to string:
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go#L658 
```go
	for idx, input := range spec.Domain.Devices.Inputs {
		if input.Bus != v1.InputBusVirtio && input.Bus != v1.InputBusUSB && input.Bus != "" {
			causes = append(causes, metav1.StatusCause{
				Type:    metav1.CauseTypeFieldValueInvalid,
				Message: "Input device can have only virtio or usb bus.",
				Field:   field.Child("domain", "devices", "inputs").Index(idx).Child("bus").String(),
			})
		}
```
where `InputBus` is:
```go
type InputBus string

const (
	InputBusUSB    InputBus = "usb"
	InputBusVirtio InputBus = "virtio"
)
```